diff --git a/backend/Cargo.toml b/backend/Cargo.toml
index aed8428..7af7041 100644
--- a/backend/Cargo.toml
+++ b/backend/Cargo.toml
@@ -22,18 +22,18 @@ async-trait = "0.1"
 env_logger = ">=0.10, <0.12"
 futures = { version = "0.3", features = ["executor"] }
 log = "0.4.17"
-sequoia-openpgp = { version = "2", default-features = false }
+sequoia-openpgp = { version = "2", default-features = false, path = "../../sequoia/openpgp" }
 thiserror = ">=1, <3"
 tempfile = "3"
 tokio = { version = "1.19", features = [ "rt-multi-thread", "io-util", "net", "macros" ] }
 
 [target.'cfg(not(windows))'.dev-dependencies]
 # Enables a crypto backend for the tests:
-sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-nettle", "__implicit-crypto-backend-for-tests"]  }
+sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-nettle", "__implicit-crypto-backend-for-tests"], path = "../../sequoia/openpgp"  }
 
 [target.'cfg(windows)'.dev-dependencies]
 # Enables a crypto backend for the tests:
-sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-cng", "__implicit-crypto-backend-for-tests"] }
+sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-cng", "__implicit-crypto-backend-for-tests"], path = "../../sequoia/openpgp" }
 
 # Enables a crypto backend for the docs.rs generation:
 [package.metadata.docs.rs]
diff --git a/gpg-agent/Cargo.toml b/gpg-agent/Cargo.toml
index b108408..3fdd4e2 100644
--- a/gpg-agent/Cargo.toml
+++ b/gpg-agent/Cargo.toml
@@ -21,25 +21,25 @@ async-trait = "0.1"
 futures = "0.3"
 log = "0.4.17"
 openpgp-cert-d = { version = "0.3.4", default-features = false }
-sequoia-gpg-agent = { version = "0.6", default-features = false }
+sequoia-gpg-agent = { version = "0.6", default-features = false, path = "../../sequoia-gpg-agent" }
 sequoia-keystore-backend = { path = "../backend", version = "0.7" }
-sequoia-openpgp = { version = "2", default-features = false }
-sequoia-ipc = { version = "0.36", default-features = false }
+sequoia-openpgp = { version = "2", default-features = false, path = "../../sequoia/openpgp" }
+sequoia-ipc = { version = "0.36", default-features = false, path = "../../sequoia/ipc" }
 tokio = { version = "1.19", features = [ "rt-multi-thread", "io-util", "net" ] }
 
 [dev-dependencies]
 env_logger = ">=0.10, <0.12"
-sequoia-openpgp = { version = "2", default-features = false, features = ["compression"] }
+sequoia-openpgp = { version = "2", default-features = false, features = ["compression"], path = "../../sequoia/openpgp" }
 tracing = {version = "0.1", default-features = false}
 tracing-subscriber = {version = "0.3", default-features = false, features = ["env-filter"]}
 
 [target.'cfg(not(windows))'.dev-dependencies]
 # Enables a crypto backend for the tests:
-sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-nettle", "__implicit-crypto-backend-for-tests"]  }
+sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-nettle", "__implicit-crypto-backend-for-tests"], path = "../../sequoia/openpgp"  }
 
 [target.'cfg(windows)'.dev-dependencies]
 # Enables a crypto backend for the tests:
-sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-cng", "__implicit-crypto-backend-for-tests"] }
+sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-cng", "__implicit-crypto-backend-for-tests"], path = "../../sequoia/openpgp" }
 
 # Enables a crypto backend for the docs.rs generation:
 [package.metadata.docs.rs]
diff --git a/keystore/Cargo.toml b/keystore/Cargo.toml
index a28f5a2..3f726e1 100644
--- a/keystore/Cargo.toml
+++ b/keystore/Cargo.toml
@@ -38,13 +38,13 @@ env_logger = ">=0.10, <0.12"
 log = "0.4.17"
 paste = "1"
 sequoia-directories = "0.1"
-sequoia-ipc = { version = "0.36", default-features = false }
+sequoia-ipc = { version = "0.36", default-features = false, path = "../../sequoia/ipc" }
 sequoia-keystore-backend = { path = "../backend", version = "0.7" }
 sequoia-keystore-softkeys = { path = "../softkeys", version = "0.7", optional = true }
 sequoia-keystore-gpg-agent = { path = "../gpg-agent", version = "0.5", optional = true }
 sequoia-keystore-openpgp-card = { path = "../openpgp-card", version = "0.2", optional = true }
 sequoia-keystore-tpm = { path = "../tpm", version = "0.2", optional = true }
-sequoia-openpgp = { version = "2", default-features = false }
+sequoia-openpgp = { version = "2", default-features = false, path = "../../sequoia/openpgp" }
 thiserror = ">=1, <3"
 tokio = { version = "1.21", features = ["rt", "net", "io-std", "macros"] }
 tokio-util = { version = "0.7", features = ["compat"] }
@@ -61,11 +61,11 @@ dircpy = "0.3"
 
 [target.'cfg(not(windows))'.dev-dependencies]
 # Enables a crypto backend for the tests:
-sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-nettle", "__implicit-crypto-backend-for-tests"]  }
+sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-nettle", "__implicit-crypto-backend-for-tests"], path = "../../sequoia/openpgp"  }
 
 [target.'cfg(windows)'.dev-dependencies]
 # Enables a crypto backend for the tests:
-sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-cng", "__implicit-crypto-backend-for-tests"] }
+sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-cng", "__implicit-crypto-backend-for-tests"], path = "../../sequoia/openpgp" }
 
 # Enables a crypto backend for the docs.rs generation:
 [package.metadata.docs.rs]
diff --git a/softkeys/Cargo.toml b/softkeys/Cargo.toml
index 3049f55..f0b0353 100644
--- a/softkeys/Cargo.toml
+++ b/softkeys/Cargo.toml
@@ -22,12 +22,12 @@ async-trait = "0.1"
 dirs = ">=5, <7"
 futures = "0.3"
 log = "0.4.17"
-sequoia-openpgp = { version = "2", default-features = false }
+sequoia-openpgp = { version = "2", default-features = false, path = "../../sequoia/openpgp" }
 
 [dev-dependencies]
 dircpy = "0.3"
 env_logger = ">=0.10, <0.12"
-sequoia-openpgp = { version = "2", default-features = false, features = ["compression"] }
+sequoia-openpgp = { version = "2", default-features = false, features = ["compression"], path = "../../sequoia/openpgp" }
 tempfile = "3"
 tokio = { version = "1.19", features = [ "rt-multi-thread", "io-util", "net" ] }
 tracing = {version = "0.1", default-features = false}
@@ -35,11 +35,11 @@ tracing-subscriber = {version = "0.3", default-features = false, features = ["en
 
 [target.'cfg(not(windows))'.dev-dependencies]
 # Enables a crypto backend for the tests:
-sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-nettle", "__implicit-crypto-backend-for-tests"]  }
+sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-nettle", "__implicit-crypto-backend-for-tests"], path = "../../sequoia/openpgp"  }
 
 [target.'cfg(windows)'.dev-dependencies]
 # Enables a crypto backend for the tests:
-sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-cng", "__implicit-crypto-backend-for-tests"] }
+sequoia-openpgp = { version = "2", default-features = false, features = ["crypto-cng", "__implicit-crypto-backend-for-tests"], path = "../../sequoia/openpgp" }
 
 # Enables a crypto backend for the docs.rs generation:
 [package.metadata.docs.rs]
